### PR TITLE
Allow weighted scalar reporting by providing a tuple to the reporter

### PR DESCRIPTION
**Pull request by mitmul**
This PR takes over #4031

### CODE REVIEW

File: chainer/reporter.py [hunk 1]
```diff
@@ -326,15 +329,25 @@ def add(self, d):
...
 
         """
         summaries = self._summaries
         for k, v in six.iteritems(d):
+            w = 1
+            if isinstance(v, tuple):
+                w = v[1]
+                v = v[0]
+                if isinstance(w, variable.Variable):
+                    w = w.data
```
Comment (toslunar) [inline chainer/reporter.py]: Use `.array` for the consistency.

**Comment by mitmul** [comment]
@okuta I added some tests. Could you check them again?
@toslunar I fixed it. Thanks!

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by okuta** [comment]
Please remove cache file.
.pytest_cache/v/cache/nodeids

**Comment by mitmul** [comment]
Removed.

Jenkins, test this please again

**Comment by okuta** [comment]
LGTM!

**Pull request merged**